### PR TITLE
[SPARK-29417][CORE] Resource Scheduling - add TaskContext.resource java api

### PR DESCRIPTION
**Pull request by tgravescs**
### What changes were proposed in this pull request?
We added a TaskContext.resources() api, but I realized this is returning a scala Map which is not ideal for access from Java.  Here I add a resourcesJMap function which returns a java.util.Map to make it easily accessible from Java.

### Why are the changes needed?
Java API access

### Does this PR introduce any user-facing change?

Yes, new TaskContext function to access from Java

### How was this patch tested?

new unit test

### CODE REVIEW

**Comment by tgravescs** [comment]
@mengxr @jiangxb1987

**Review by jiangxb1987 — APPROVED** [review]
LGTM, pending jenkins

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Retest this please.

**Comment by dongjoon-hyun** [comment]
Retest this please.

File: core/src/main/scala/org/apache/spark/TaskContext.scala [hunk 1]
```diff
@@ -185,6 +185,14 @@ abstract class TaskContext extends Serializable {
   @Evolving
   def resources(): Map[String, ResourceInformation]
 
+  /**
+   * (java-specific) Resources allocated to the task. The key is the resource name and the value
+   * is information about the resource. Please refer to
+   * [[org.apache.spark.resource.ResourceInformation]] for specifics.
+   */
+  @Evolving
+  def resourcesJMap(): java.util.Map[String, ResourceInformation]
```
Comment (dongjoon-hyun) [inline core/src/main/scala/org/apache/spark/TaskContext.scala]: Ah, I missed that. We need MiMa exclusion.
```
[error]  * abstract method resourcesJMap()java.util.Map in class org.apache.spark.TaskContext is present only in current version
[error]    filter with: ProblemFilters.exclude[ReversedMissingMethodProblem]("org.apache.spark.TaskContext.resourcesJMap")
```

Comment (tgravescs) [reply core/src/main/scala/org/apache/spark/TaskContext.scala]: I'll update shortly

**Comment by mengxr** [comment]
Merged into master. Thanks!

**Comment by dongjoon-hyun** [comment]
Thank you, all!

**Pull request closed without merging**